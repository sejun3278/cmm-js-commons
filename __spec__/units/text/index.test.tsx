import React from "react";

import { render } from "@testing-library/react";
import _Text from "../../../src/units/text";

export default describe("Text Page", () => {
  // 스냅샷
  test("Snapshot - span", () => {
    const { container } = render(<_Text>span 태그를 렌더합니다.</_Text>);
    expect(container).toMatchSnapshot();
  });

  // p태그 렌더 스냅샷
  test("Snapshot - p", () => {
    const { container } = render(<_Text type="p">p 태그를 렌더합니다.</_Text>);
    expect(container).toMatchSnapshot();
  });

  // span 태그 존재여부와 클래스, 텍스트 값 확인하기
  test("Check Text and class", () => {
    const { container } = render(
      <_Text type="span" className="jest-span">
        span 태그를 렌더합니다.
      </_Text>
    );

    const spanEle = container.querySelector(".jest-span");
    if (spanEle) {
      // span 태그가 존재하는지 검증
      expect(spanEle).toBeInTheDocument();
      // 클래스 여부 검증
      expect(spanEle).toHaveClass("jest-span mcm-span-unit");
      // 텍스트 여부 검증
      expect(spanEle.textContent).toEqual("span 태그를 렌더합니다.");
    }
  });
});
