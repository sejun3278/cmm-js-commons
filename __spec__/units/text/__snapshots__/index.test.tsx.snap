// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Page Snapshot - p 1`] = `
<div>
  <p
    class="mcm-p-unit css-pzlb5d"
  >
    p 태그를 렌더합니다.
  </p>
</div>
`;

exports[`Text Page Snapshot - span 1`] = `
<div>
  <span
    class="mcm-span-unit"
  >
    span 태그를 렌더합니다.
  </span>
</div>
`;

exports[`Text Page Snapshot 1`] = `
<div>
  <span
    class="mcm-span-unit"
  >
    span 태그를 렌더합니다.
  </span>
</div>
`;

exports[`Text Page Snapshot 2`] = `
<div>
  <p
    class="mcm-p-unit css-pzlb5d"
  >
    p 태그를 렌더합니다.
  </p>
</div>
`;
